src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,179 @@
+        String sql = "SELECT * FROM info where id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = parseRow(resultSet);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not connect to DB");
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM info;";
+        List<Book> result = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery(sql);
+            while (resultSet.next()) {
+                Book book = parseRow(resultSet);
+                result.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not connect to the DB");
+        }
+        return result;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE info SET book_name = ?, author_name = ?, price = ?,"
+                + " quantity = ?, published_year = ? WHERE id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+
+            statement.setString(1, book.getName());
+            statement.setString(2, book.getAuthor());
+            statement.setBigDecimal(3, book.getPrice());
+            statement.setLong(4, book.getQuantity());
+            statement.setInt(5, book.getYear());
+            statement.setLong(6, book.getId());
+
+            int affectedRows = statement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new DataProcessingException("No rows updated. Book with id "
+                        + book.getId() + " not found.");
+            }
+
+            return book;
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot update book in the DB");
+        }
+    }
+
+    @Override
+    public boolean delete(Book book) {
+        String sql = "DELETE FROM info WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, book.getId());
+
+            int affectedRows = statement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to delete at least one row,"
+                        + " but deleted 0 rows.");
+            }
```

Comment:
remove unnecessary code
```suggestion
            return statement.executeUpdate() > 0;
```